subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -4643,176 +4645,150 @@ static void mayfly_xtal_stop_calc(void *params)
+
+				ticker_status = ticker_update(
+					RADIO_TICKER_INSTANCE_ID_RADIO,
+					RADIO_TICKER_USER_ID_JOB,
+					ticker_id,
+					ticks_drift_plus, 0,
+					0, ticks_drift_plus,
+					0, 0,
+					prepare_reduced,
+					hdr);
+				LL_ASSERT((TICKER_STATUS_SUCCESS ==
+					   ticker_status) ||
+					  (TICKER_STATUS_BUSY ==
+					   ticker_status));
 			}
+		}
 
 #if defined(CONFIG_BT_CTLR_SCHED_ADVANCED)
-			if (ticker_id_current >= RADIO_TICKER_ID_FIRST_CONNECTION) {
-				/* compensate the current ticker for reduced
-				 * prepare.
-				 */
-				conn_curr =
-					mem_get(_radio.conn_pool,
-						CONNECTION_T_SIZE,
-						(ticker_id_current -
-						 RADIO_TICKER_ID_FIRST_CONNECTION));
-				ticks_prepare_to_start =
-					(conn_curr->hdr.ticks_active_to_start >
-					conn_curr->hdr.ticks_preempt_to_start) ?
-					conn_curr->hdr.ticks_active_to_start :
-					conn_curr->hdr.ticks_preempt_to_start;
-				if ((conn_curr->hdr.ticks_xtal_to_start &
-						((u32_t)1 << 31)) != 0) {
-					ticks_to_expire +=
-						((conn_curr->hdr.ticks_xtal_to_start &
-						  (~((u32_t)1 << 31))) -
-						 ticks_prepare_to_start);
-				}
-			}
+		if (!conn) {
+			return;
+		}
 
-			/* auto conn param req or conn update procedure to
-			 * avoid connection collisions.
+		if (ticker_id_current >= RADIO_TICKER_ID_FIRST_CONNECTION) {
+			/* compensate the current ticker for reduced
+			 * prepare.
 			 */
-			if ((conn) && (conn_curr) &&
-			    (conn_curr->conn_interval == conn->conn_interval)) {
-				u32_t ticks_conn_interval =
-					TICKER_US_TO_TICKS(conn->conn_interval * 1250);
-
-				/* remove laziness, if any, from
-				 * ticks_to_expire.
-				 */
-				while (ticks_to_expire > ticks_conn_interval) {
-					ticks_to_expire -= ticks_conn_interval;
-				}
-
-				/* if next ticker close to this ticker, send
-				 * conn param req.
-				 */
-				if (conn_curr->role && !conn->role &&
-				    (ticks_to_expire <
-				     (TICKER_US_TO_TICKS(RADIO_TICKER_XTAL_OFFSET_US +
-							 625) +
-				      conn_curr->hdr.ticks_slot))) {
-					u32_t status;
-
-					status = conn_update_req(conn_curr);
-					if ((status == 2) &&
-					    (conn->llcp_version.rx)) {
-						conn_update_req(conn);
-					}
-				} else if (!conn_curr->role && conn->role &&
-					   (ticks_to_expire <
-					    (TICKER_US_TO_TICKS(RADIO_TICKER_XTAL_OFFSET_US +
-								625) +
-					     conn_curr->hdr.ticks_slot))) {
-					u32_t status;
-
-					status = conn_update_req(conn);
-					if ((status == 2) &&
-					    (conn_curr->llcp_version.rx)) {
-						conn_update_req(conn_curr);
-					}
-				}
+			conn_curr = mem_get(_radio.conn_pool, CONNECTION_T_SIZE,
+					    (ticker_id_current -
+					     RADIO_TICKER_ID_FIRST_CONNECTION));
+			ticks_prepare_to_start =
+				(conn_curr->hdr.ticks_active_to_start >
+				conn_curr->hdr.ticks_preempt_to_start) ?
+				conn_curr->hdr.ticks_active_to_start :
+				conn_curr->hdr.ticks_preempt_to_start;
+			if (conn_curr->hdr.ticks_xtal_to_start & BIT(31)) {
```

Comment:
Should this be `BIT(23)` if it's int RTC ticks?

Reply:
This is special bit to differentiate a normal preparation duration v/s a preemption with xtal already running.

Reply:
OK, it would be good to use a macro here for this instead of just `31` but this can be done on another patch

Reply:
Yes, trying to clean up, more will come, one small step at a time.